src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -56,14 +58,41 @@ public static class TaskExtensions {
         }
 
         /// <summary>
-        /// Suppresses warnings about unawaited tasks and ensures that unhandled
-        /// errors will cause the process to terminate.
+        /// Rethrows task exceptions back to the callers synchronization context if it is possible
         /// </summary>
         /// <remarks>
         /// <see cref="OperationCanceledException"/> is always ignored.
```

Comment:
When task failed with OCE or `taskCompletionSource.SetCanceled()` is called, `Task.IsFaulted` will be `false`.
The only case when it is faulted is when you explicitly set `OperationCanceledException`:
```{csharp}
taskCompletionSource.SetException(new OperationCanceledException())
```
but that is a wrong way to cancel `TaskCompletionSource` and I'd prefer to fail for such code.